File: backend/RS1_2024_25.API/Data/Models/Ticket.cs
using RS1_2024_25.API.Data.Models.Auth;
using RS1_2024_25.API.Data.Models;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace RS1_2024_25.API.Data
{
    public class Ticket
    {
        [Key]
        public required int Id { get; set; }
        [ForeignKey(nameof(MyAppUser))]
        public required int UserId { get; set; }
        public MyAppUser? User { get; set; }
        [ForeignKey(nameof(Line))]
        public required int LineId { get; set; }
        public Line? Line { get; set; }
        [ForeignKey(nameof(Zone))]
        public required int ZoneId { get; set; }
        public Zone? Zone { get; set; }
        [ForeignKey(nameof(TicketType))]
        public required int TicketTypeId { get; set; }
        public TicketType? TicketType { get; set; }
        public required DateTime CreatedDate { get; set; }
        public required DateTime ExpirationDate { get; set; }
        public required byte[] QrCode { get; set; }
    }
}

File: Frontend/Client/src/app/_models/ticket.ts
import { Line } from './line';
import { MyAppUser } from './my-app-user';
import { TicketInfo } from './ticket-info';
import { TicketType } from './ticket-type';
import { Zone } from './zone';

export interface Ticket {
  id: number;
  userId: number;
  ticketInfoId: number;
  ticketInfo: TicketInfo;
  createdDate: string;
  expirationDate: string;
  qrCode: string;
}
